### PR TITLE
Add an option to retrieve schema files securely/insecurely

### PR DESCRIPTION
**Pull request by xuhdev**
This defaults to require a secure link (https with a valid TLS
certificate), but a user has the option to lower the requirements.

Also increase the use of https links instead of http links for our
internal testing purposes, otherwise they would fail for insecure
connection exceptions.

---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Close #internal

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [x] When the pull request is ready, request review.
</details>

### CODE REVIEW

File: pydax/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"Exceptions of this module."
+
+
+class InsecureConnectionError(RuntimeError):
```
Comment (edwardleardi) [inline pydax/exceptions.py:20]: Do we need to document this since its public?

File: tests/test_schema_retrieval.py [hunk 2]
```diff
@@ -16,40 +16,45 @@
...
+from pydax.exceptions import InsecureConnectionError
 from pydax._schema_retrieval import retrieve_schema_file
 
 
 class TestSchemaRetrieval:
     "Test schema retrieval."
 
+    # We set tls_verification to False in this test. We don't test raising this warning because it is not within the
+    # project's scope to guarantee that this warning will raise, and this warning raises only once, which means that
+    # whether the warning will be raised depending on the order in which the tests being executed.
```
Comment (edwardleardi) [inline tests/test_schema_retrieval.py]:
```suggestion
    # whether the warning will be raised depends on the order in which the tests are being executed.
```

**Review by edwardleardi — COMMENTED** [review]
LGTM, couple of minor things

File: pydax/exceptions.py (hunk 1 above)
Comment (xuhdev) [reply pydax/exceptions.py:20]: Yes, it is documented. What do you mean?

Comment (xuhdev) [reply pydax/exceptions.py:20]: I see what you mean. I would refer this to `retrieve_schema_file`, but it is private and we haven't reached the step of dealing with a public function yet. When a public function on security setup is made, we can refer to that document for when this exception is thrown.

Comment (edwardleardi) [reply pydax/exceptions.py:20]: oh ok gotcha

**Review by edwardleardi — APPROVED** [review]

**Pull request merged**